notebooks/GWAS_1000Genomes_p1/main.py

```diff
@@ -0,0 +1,143 @@
+
+    ### Install packages
+    """
+
+with herzog.Cell("python"):
+    #%pip install tenacity
+    pass
+
+with herzog.Cell("markdown"):
+    """
+    Make sure to restart the kernel whenever you pip install things in a Jupyter notebook.
+
+    ### Import all the packages this notebook will use
+    """
+
+with herzog.Cell("python"):
+    from firecloud import fiss
+    import tenacity
+    import pandas as pd
+    pd.set_option('display.max_row', 10)
+    import os
+    import io
+    import numpy as np
+    import seaborn as sns
+
+with herzog.Cell("markdown"):
+    """
+    ## Define filepaths and environmental variables
+    """
+
+with herzog.Cell("python"):
+    PROJECT = os.environ['GOOGLE_PROJECT']
+    PROJECT
+
+with herzog.Cell("python"):
+    WORKSPACE = os.environ['WORKSPACE_NAME']
+    WORKSPACE
+
+with herzog.Cell("python"):
+    bucket = os.environ['WORKSPACE_BUCKET']
+    bucket = bucket + '/'
+    bucket
+
+with herzog.Cell("markdown"):
+    """
+    Consolidate the Gen3 clinical data into a single Terra data table using functions in the "terra_data_table_util" python notebook
+    """
+
+with herzog.Cell("python"):
+    # Run the companion notebook. Note: it must be in the same workspace you're currently working in.
+    #%run terra_data_table_util.ipynb
+    pass
+
+with herzog.Cell("python"):
+    # Take a look at all the entities (tables) in the workspace
+    ent_types = fiss.fapi.list_entity_types(PROJECT, WORKSPACE).json()
+    for t in ent_types.keys():
+        print(t, "count:", ent_types[t]['count'])
+
+with herzog.Cell("python"):
+    # Set the name for your consolidated table
+    consolidated_table_name = "consolidated_metadata"
+
+with herzog.Cell("markdown"):
+    """
+    The consolidate_gen3_pheno_tables function:
+    * Joins all clinical data tables into a single consolidated_metadata table in the Terra data section
+    * This join forces all entities (individuals) to be present in every clinical table, so some individuals may be removed. Consider how this affects your dataset.
+    * Renames attribute fields to have a prefix of the original entity type (for example: "demographic_annotated_sex", where demographic is the original entity type, annotated_sex is the attribute field)
+
+    When working on data imported from Gen3, you will see most columns have names beginning with "pfb:" before the rest of the column name such as pfb:demographic_annotated_sex. PFB, which stands for Portable Format for Bioinfomatics, is a common namespace attribute that aids in interoperability with other NIH ecosystems.
+    """
+
+with herzog.Cell("python"):
+    # Consolidate the phenotypic data using the function defined in the terra_data_table_util notebook
+    consolidate_gen3_pheno_tables(PROJECT, WORKSPACE, consolidated_table_name)  #type: ignore  # noqa
```

Comment:
What is #type ignore #noqa for?

Reply:
This was previously necessary to get the notebooks to pass on our gitlab pipeline (flake8 to be precise), but since this function is now being mocked, these marks are not needed anymore. Thanks for spotting that.

FYI the second notebook has a few lines that are marked with that that cannot be removed as far as I'm aware, but they should not be in any of the functions that are actually seen by the user -- ie, they're only in stuff outside of herzog cells so they won't appear in the final notebook.